src/master-playlist-controller.js

```diff
@@ -684,8 +684,16 @@ export class MasterPlaylistController extends videojs.EventTarget {
       }
     }
 
-    if (isEndOfStream) {
+    if (!isEndOfStream) {
+      return;
+    }
+
+    // on chrome calling endOfStream can sometimes cause an exception,
```

Comment:
Chrome can sometimes throw a DOM exception here when `endOfStream` is called. It seems like it almost always happens when:
1. `sourceBuffers` are still updating
2. The `mediaSource` is no longer `open`

It does seem to happen at other times though. So I think we just need to wrap this in a try catch here.